src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -444,55 +475,67 @@ private AIState clearStep()
 
     private AIState requestMaterials()
     {
-        if(job.getStructure() == null)
+        if (job.getStructure() == null)
         {
             //fix for bad structures
             job.complete();
         }
 
-        //todo as soon as material handling has been implemented this should be set to work!
         //We need to deal with materials
-        if (!Configurations.builderInfiniteResources)
+        if (!Configurations.builderInfiniteResources && !job.getWorkOrder().isRequested())
         {
-            //TODO thread this
             while (job.getStructure().findNextBlock())
             {
                 if (job.getStructure().doesStructureBlockEqualWorldBlock())
                 {
                     continue;
                 }
 
-                @Nullable Block block = job.getStructure().getBlock();
-                @NotNull ItemStack itemstack = new ItemStack(block, 1);
+                @Nullable final Block block = job.getStructure().getBlock();
+
+                @NotNull final ItemStack itemstack = new ItemStack(block, 1);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "itemstack" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "itemstack". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1854)